src/Proto3/Suite/Class.hs

```diff
@@ -175,7 +177,7 @@ instance (Bounded e, Enum e) => HasDefault (Enumerated e) where
     case toEnumMay 0 of
       Nothing -> Enumerated (Left 0)
       Just x -> Enumerated (Right x)
-  isDefault = (== 0) . either id fromEnum . enumerated
+  isDefault = (== 0) . either id fromEnum . coerce @(Enumerated e) @(Either Int e)
```

Comment:
In this case I prefer the old syntax; the efficiency should be the same.

Reply:
Ok, I verified that the simplifier output is the same for both versions

Reply:
reverted in  9309ab1